### PR TITLE
Docs: update text and example

### PR DESCRIPTION
**Pull request by ektravel**
This PR updates the consumer properties section of the Apache Kafka ingestion topic. Specifically, it updates the dynamic config provider example and clarifies how the `isolation.level` property affects the way Druid reads transactional messages.

This PR has:

- [ x ] been self-reviewed.

### CODE REVIEW

File: docs/ingestion/kafka-ingestion.md [hunk 1]
```diff
@@ -143,41 +143,54 @@ Similarly, you can use `metrics-.*` as the value for `topicPattern` if you want
 
 #### Consumer properties
 
-Consumer properties control how a supervisor reads and processes event messages from a Kafka stream. For more information about consumers, refer to the [Apache Kafka documentation](https://kafka.apache.org/documentation/#consumerconfigs).
+Consumer properties control how a supervisor reads and processes event messages from a Kafka stream. For more information about consumer configs and advanced use cases, refer to the [Kafka documentation](https://kafka.apache.org/documentation/#consumerconfigs).
 
-The `consumerProperties` object must contain a  `bootstrap.servers` property with a list of Kafka brokers in the form: `<BROKER_1>:<PORT_1>,<BROKER_2>:<PORT_2>,...`.
-By default, `isolation.level` is set to `read_committed`.
+Consumer properties must include a `bootstrap.servers` property with a list of Kafka brokers in the format `<BROKER_1>:<PORT_1>,<BROKER_2>:<PORT_2>,...`.
```
Comment (writer-jill) [inline docs/ingestion/kafka-ingestion.md:148]:
```suggestion
You must include `bootstrap.servers` in consumer properties with a list of Kafka brokers in the format `<BROKER_1>:<PORT_1>,<BROKER_2>:<PORT_2>,...`.
```

File: docs/ingestion/kafka-ingestion.md [hunk 2]
```diff
@@ -143,41 +143,54 @@ Similarly, you can use `metrics-.*` as the value for `topicPattern` if you want
...
 
-If you use older versions of Kafka servers without transactions support or don't want Druid to consume only committed transactions, set `isolation.level` to `read_uncommitted`. If you need Druid to consume older versions of Kafka, make sure offsets are sequential, since there is no offset gap check in Druid.
+The `isolation.level` property in `consumerProperties` determines how Druid reads messages written transactionally.
+If you use older versions of Kafka servers without transaction support or don't want Druid to consume only committed transactions, set `isolation.level` to `read_uncommitted`.
```
Comment (writer-jill) [inline docs/ingestion/kafka-ingestion.md:151]:
```suggestion
If you use older versions of Kafka servers without transaction support or you don't want Druid to consume only committed transactions, set `isolation.level` to `read_uncommitted`.
```

File: docs/ingestion/kafka-ingestion.md [hunk 3]
```diff
@@ -143,41 +143,54 @@ Similarly, you can use `metrics-.*` as the value for `topicPattern` if you want
...
+In some cases, you may need to retrieve consumer properties at runtime. For example, when `bootstrap.servers` is unknown or not static. To enable SSL connections, you must provide passwords for `keystore`, `truststore`, and `key` confidentially. You can specify these settings in the `jaas.conf` login configuration file or in `consumerProperties` with `sasl.jaas.config`.
 
-For example, if you are using SASL and SSL with Kafka, set the following environment variables for the Druid user on the machines running the Overlord and the Peon services:
+To protect sensitive information, use the [environment variable dynamic config provider](../operations/dynamic-config-provider.md#environment-variable-dynamic-config-provider) to store credentials in system environment variables instead of plain text.
+You can also use the [password provider](../operations/password-provider.md) interface to set `keystore`, `truststore`, and `key` properties, but this feature is deprecated.
```
Comment (writer-jill) [inline docs/ingestion/kafka-ingestion.md:161]: Should we avoid recommending deprecated features?

**Review by writer-jill — APPROVED** [review]
A couple of comments, looks good!

**Comment by daveklein** [comment]
I reviewed this from a Kafka user perspective and it looks good to me.

File: docs/ingestion/kafka-ingestion.md [hunk 4]
```diff
@@ -143,41 +143,54 @@ Similarly, you can use `metrics-.*` as the value for `topicPattern` if you want
 
 #### Consumer properties
 
-Consumer properties control how a supervisor reads and processes event messages from a Kafka stream. For more information about consumers, refer to the [Apache Kafka documentation](https://kafka.apache.org/documentation/#consumerconfigs).
+Consumer properties control how a supervisor reads and processes event messages from a Kafka stream. For more information about consumer configs and advanced use cases, refer to the [Kafka documentation](https://kafka.apache.org/documentation/#consumerconfigs).
```
Comment (techdocsmith) [inline docs/ingestion/kafka-ingestion.md:146]:
```suggestion
Consumer properties control how a supervisor reads and processes event messages from a Kafka stream. For more information about consumer configuration and advanced use cases, refer to the [Kafka documentation](https://kafka.apache.org/documentation/#consumerconfigs).
```
nit

File: docs/ingestion/kafka-ingestion.md [hunk 5]
```diff
@@ -143,41 +143,54 @@ Similarly, you can use `metrics-.*` as the value for `topicPattern` if you want
...
+In some cases, you may need to retrieve consumer properties at runtime. For example, when `bootstrap.servers` is unknown or not static. To enable SSL connections, you must provide passwords for `keystore`, `truststore`, and `key` confidentially. You can specify these settings in the `jaas.conf` login configuration file or in `consumerProperties` with `sasl.jaas.config`.
 
-For example, if you are using SASL and SSL with Kafka, set the following environment variables for the Druid user on the machines running the Overlord and the Peon services:
+To protect sensitive information, use the [environment variable dynamic config provider](../operations/dynamic-config-provider.md#environment-variable-dynamic-config-provider) to store credentials in system environment variables instead of plain text.
+Although you can also use the [password provider](../operations/password-provider.md) interface to specify TLS configs for Kafka ingestion, consider using the dynamic config provider as this feature is deprecated.
```
[the remark this answers is not in the document]
Comment (ektravel) [reply docs/ingestion/kafka-ingestion.md]: Updated

**Review by techdocsmith — APPROVED** [review]
LGTM

**Pull request merged**